### PR TITLE
General code clean up and Feature 1 work

### PR DESCRIPTION
**Pull request by scheffeltravis**
This PR encompasses a lot of refactoring and clean up but the main things to note here are:
- Combined the search and recipe templates to simplify things
- Wired up searching to recipe database
- Added simple category search (which should probably just be a dropdown but we can do that later)

Here are some screenshots of it in action (because if I'm honest I don't know how to do a video:
1. When first logging in (you will see the edit button is only available for the logged in user):
image

2. Searching by title:
image

3. Searching by category:
image
image

4. Searching by author:
image
image

The other thing to note is we don't save state in anyway so when the search POST request is made, the page reloads with the radio buttons reset.

### CODE REVIEW

File: example/flaskr/templates/recipe/create.html [hunk 1]
```diff
@@ -10,6 +10,14 @@ <h1>{% block title %}New Post{% endblock %}</h1>
     <input name="title" id="title" value="{{ request.form['title'] }}" required>
     <label for="body">Body</label>
     <textarea name="body" id="body">{{ request.form['body'] }}</textarea>
+    <label for="category">Recipe Category</label>
+    <select id="category" name="category">
+      <option value="1">Breakfast</option>
```
Comment (CHRYSICS) [inline example/flaskr/templates/recipe/create.html:15]: This is nicely done, like how it this works well for the search

**Review by CHRYSICS — APPROVED** [review]
Looks good to me and runs well!

File: example/flaskr/templates/recipe/index.html [hunk 2]
```diff
@@ -1,28 +1,51 @@
...
+  {% for recipe in recipes %}
+    <article class="recipe">
+      <header>
         <div>
-            <h1>{{ recipe['title'] }}</h1>
-            <div class="about">by {{ recipe['username'] }} on {{ recipe['created'].strftime('%Y-%m-%d') }}</div>
+          <h1>{{ recipe['title'] }}</h1>
+          <div class="about">by {{ recipe['username'] }} on {{ recipe['created'].strftime('%Y-%m-%d') }}</div>
         </div>
-
```
Comment (CHRYSICS) [inline example/flaskr/templates/recipe/index.html:18]: This is the only thing I would leave actually. The select will bring us to a view template where edit and save to recipe book should be there.  In F1-01, one requirement is to be able to view recipe

Comment (scheffeltravis) [reply example/flaskr/templates/recipe/index.html:18]: AH, good catch. I actually think this got cleaned up when I was dealing with merge conflicts. I'll fix it up

Comment (CHRYSICS) [reply example/flaskr/templates/recipe/index.html:18]: Just committed, should have fixed it :+1:

**Review by CHRYSICS — APPROVED** [review]

**Pull request merged**